packages/placement-ordering/configure/src/choice-editor.jsx

```diff
@@ -21,33 +20,80 @@ function findFreeChoiceSlot(choices) {
 
 const log = debug('@pie-element:placement-ordering:configure:choice-editor');
 
+function updateResponse(response, from, to) {
```

Comment:
this looks very similar to some pre-existing logic we have - we can't reuse?

Reply:
There is a similar function is pie-ui
function updateResponse(state, from, to) {
  const { response, opts } = state;
  const update = cloneDeep(response);
  if (opts.includeTargets) {
    if (from.type === 'choice' && to.type === 'target') {
      update[to.index] = from.id;
      return update;
    } else if (from.type === 'target' && to.type === 'choice') {
      update[from.index] = undefined;
      return update;
    } else if (from.type === 'target' && to.type === 'target') {
      return swap(response, from.index, to.index);
    } else {
      log('do nothing to the response');
      return response;
    }
  } else {
    const fromIndex = state.response.findIndex(
      r => r !== undefined && r === from.id
    );
    const toIndex = state.response.findIndex(
      r => r !== undefined && r === to.id
    );
    log('fromIndex: ', fromIndex, 'toIndex:', toIndex);
    return swap(state.response, fromIndex, toIndex);
  }
}
but it works a little bit different and returns a different value.
I believe customising it will make it pretty complex.
What do you think?

Reply:
parked for now